relay-redis/src/real.rs

```diff
@@ -382,3 +396,154 @@ pub struct Stats {
     /// The number of idle connections.
     pub idle_connections: u32,
 }
+
+/// [`bb8::ManageConnection`] for a async redis cluster.
+pub struct RedisClusterConnectionManager {
+    client: ClusterClient,
+}
+
+impl RedisClusterConnectionManager {
+    /// Creates a new [`bb8::ManageConnection`] with the provided [`ClusterClient`]
+    pub fn new(client: ClusterClient) -> Self {
+        Self { client }
+    }
+}
+
+impl bb8::ManageConnection for RedisClusterConnectionManager {
+    type Connection = ClusterConnection;
+    type Error = redis::RedisError;
+
+    fn connect<'life0, 'async_trait>(
+        &'life0 self,
+    ) -> Pin<Box<dyn Future<Output = Result<Self::Connection, Self::Error>> + Send + 'async_trait>>
+    where
+        'life0: 'async_trait,
+    {
+        let client = self.client.clone();
+        Box::pin(async move { client.get_async_connection().await })
+    }
+
+    fn is_valid<'life0, 'life1, 'async_trait>(
+        &'life0 self,
+        conn: &'life1 mut Self::Connection,
+    ) -> Pin<Box<dyn Future<Output = Result<(), Self::Error>> + Send + 'async_trait>>
+    where
+        'life0: 'async_trait,
+        'life1: 'async_trait,
+    {
+        Box::pin(async move { redis::cmd("PING").query_async(conn).await })
```

Comment:
Should also check the response if it is `PONG`.

See also: https://docs.rs/bb8-redis/0.17.0/src/bb8_redis/lib.rs.html#47-49